packages/eslint-plugin/src/rules/space-before-blocks.ts

```diff
@@ -0,0 +1,102 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+import { getESLintCoreRule } from '../util/getESLintCoreRule';
+import * as util from '../util';
+
+const baseRule = getESLintCoreRule('space-before-blocks');
+
+export type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
+export type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
+
+export default util.createRule<Options, MessageIds>({
+  name: 'space-before-blocks',
+  meta: {
+    type: 'layout',
+    docs: {
+      description: 'Enforces consistent spacing before blocks',
+      recommended: false,
+      extendsBaseRule: true,
+    },
+    fixable: baseRule.meta.fixable,
+    hasSuggestions: baseRule.meta.hasSuggestions,
+    schema: baseRule.meta.schema,
+    messages: {
+      // @ts-expect-error -- we report on this messageId so we need to ensure it's there in case ESLint changes in future
+      unexpectedSpace: 'Unexpected space before opening brace.',
+      // @ts-expect-error -- we report on this messageId so we need to ensure it's there in case ESLint changes in future
+      missingSpace: 'Missing space before opening brace.',
+      ...baseRule.meta.messages,
+    },
+  },
+  defaultOptions: ['always'],
+  create(context) {
+    const rules = baseRule.create(context);
+    const config = context.options[0];
+    const sourceCode = context.getSourceCode();
+
+    let alwaysClasses = true,
+      neverClasses = false;
+
+    if (typeof config === 'object') {
+      alwaysClasses = config.classes === 'always';
+      neverClasses = config.classes === 'never';
+    } else if (config === 'never') {
+      alwaysClasses = false;
+      neverClasses = true;
+    }
+
+    function checkPrecedingSpace(
+      node:
+        | TSESTree.Token
+        | TSESTree.TSInterfaceBody
+        | TSESTree.PunctuatorToken,
+    ): void {
+      const precedingToken = sourceCode.getTokenBefore(node);
+      if (
+        precedingToken &&
+        util.isTokenOnSameLine(precedingToken, node as TSESTree.Token)
+      ) {
+        const hasSpace = sourceCode.isSpaceBetweenTokens(
```

Comment:
Err, `isSpaceBetweenTokens` is marked as `@deprecated in favor of isSpaceBetween`. Have you tried using that instead?

Reply:
Replaced with `isSpaceBetween`, it is marked as optional though and available since 6.7. Should I use both then?
New method also doesn't require explicit cast, so that is nice.

Reply:
Ah sorry for the delay @FDIM, missed your reply till now! Looking a bit closer, yes, `eslint-plugin` supports `"eslint": "^5.0.0 || ^6.0.0 || ^7.0.0"`. So we'll have to support both I think.